extension/content/components/recipes/form/arguments/PreferenceExperimentArguments.js

```diff
@@ -0,0 +1,393 @@
+          <Col xs={24}>
+            <Branches />
+          </Col>
+        </Row>
+      </FormGroup>
+    </>
+  );
+}
+
+function InputField({ label, name }) {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+
+  const handleChange = (value) => {
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data: {
+        ...data,
+        arguments: {
+          ...data.arguments,
+          [name]: value,
+        },
+      },
+    });
+  };
+
+  return (
+    <FormGroup>
+      <ControlLabel>{label}</ControlLabel>
+      <Input value={data.arguments[name]} onChange={handleChange} />
+    </FormGroup>
+  );
+}
+
+InputField.propTypes = {
+  label: PropTypes.string,
+  name: PropTypes.string.isRequired,
+};
+
+function ToggleField({ children, label, name }) {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+
+  const handleChange = (value) => {
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data: {
+        ...data,
+        arguments: {
+          ...data.arguments,
+          [name]: value,
+        },
+      },
+    });
+  };
+
+  return (
+    <FormGroup>
+      <ControlLabel>{label}</ControlLabel>
+      <div className="d-flex">
+        <span className="pr-2 pt-1">
+          <Toggle value={data.arguments[name]} onChange={handleChange} />
+        </span>
+        <HelpBlock className="flex-grow-1">{children}</HelpBlock>
+      </div>
+    </FormGroup>
+  );
+}
+
+ToggleField.propTypes = {
+  children: PropTypes.any,
+  label: PropTypes.string,
+  name: PropTypes.string.isRequired,
+};
+
+function SelectField({ label, name, options }) {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+
+  const handleChange = (value) => {
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data: {
+        ...data,
+        arguments: {
+          ...data.arguments,
+          [name]: value,
+        },
+      },
+    });
+  };
+
+  return (
+    <FormGroup>
+      <ControlLabel>{label}</ControlLabel>
+      <SelectPicker
+        block
+        cleanable={false}
+        data={options}
+        searchable={false}
+        value={data.arguments[name]}
+        onChange={handleChange}
+      />
+    </FormGroup>
+  );
+}
+
+SelectField.propTypes = {
+  options: PropTypes.array.isRequired,
+  label: PropTypes.string,
+  name: PropTypes.string.isRequired,
+};
+
+function Branches() {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+  const { preferenceType } = data.arguments;
+
+  React.useEffect(() => {
+    const branches = data.arguments.branches || [];
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data: {
+        ...data,
+        arguments: {
+          ...data.arguments,
+          branches: branches.map((b) => {
+            if (preferenceType === "boolean") {
+              return { ...b, value: false };
+            }
+
+            return { ...b, value: "" };
```

Comment:
Would this override displaying the actual branch values for recipes in the edit and import workflow?

Reply:
good catch, i'll fix this